projects/driver_controls/inc/pedal_calibration.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+// This module provides the means for pedal calibration.
+// Throttle module takes in a structure, ThrottleCalibrationData, that is calibrated in this module.
+//
+// Calibration has basically 3 stages. First stage is about reading and recording pedal inputs from
+// pedal at full brake position. The second stage is same as first but for full throttle position.
+// The last stage is about calculating the results into a ThrottleCalibrationData structure that
+// throttle module uses.
+//
+// The procedure is set up in test_pedal_calibration with the help of a FSM that
+// when switched to a state calls the corresponding functions from this module. Look at
+// pedal_calibration_fsm module for more information.
+#include "ads1015.h"
+#include "throttle.h"
+
+// Pedal takes 2 ADC channels.
+// Calibration determines the high and the low resolution channel.
+typedef enum {
+  PEDAL_CALIBRATION_CHANNEL_A = 0,
+  PEDAL_CALIBRATION_CHANNEL_B,
+  NUM_PEDAL_CALIBRATION_CHANNELS
+} PedalCalibrationChannel;
+
+// Pedal inputs are read in 2 positions: full throttle and full brake.
+typedef enum {
+  PEDAL_CALIBRATION_STATE_FULL_BRAKE = 0,  // Fully pressed.
+  PEDAL_CALIBRATION_STATE_FULL_THROTTLE,   // Fully depressed.
+  NUM_PEDAL_CALIBRATION_STATES
+} PedalCalibrationState;
+
+// The range is the structure that holds max and min ADC readings on a channel in a fixed state.
+// For example, reading the pedal at full brake position would give a range of values, not one.
+typedef struct PedalCalibrationRange {
+  int16_t min;
+  int16_t max;
+} PedalCalibrationRange;
+
+typedef struct PedalCalibrationSettings {
+  Ads1015Storage *ads1015_storage;
+  // This should be the final product of calibration ready to be used by throttle module.
+  ThrottleCalibrationData *throttle_calibration_data;
+  Ads1015Channel adc_channel[NUM_PEDAL_CALIBRATION_CHANNELS];
+  uint8_t brake_zone_percentage;  // [0, x]
+  uint8_t coast_zone_percentage;  // (x, y]
+  uint8_t sync_safety_factor;     // difference between primary/secondary channels - multiplier
+  uint8_t bounds_tolerance;       // percentage tolerance for min/max bounds
+} PedalCalibrationSettings;
+
+typedef struct PedalCalibrationStorage {
+  // TODO(ELEC-382): remove from this
```

Comment:
I don't like having the `ThrottleStorage` struct in `PedalCalibrationStorage` because it's really only used in validation, but moving it somewhere else would be effort. Honestly, the calibration process can probably use more thought. I think it'd be nice if we could have a single calibration program that can do all the calibration for driver controls.

Reply:
Oh I just meant that the `TODO` was ambiguous but that change makes sense. I'll wait for further revisions before adding any comments.